### DISCUSSION TITLE
‏‏‎ Firestore‏‏‎

### DISCUSSION THREAD

**Discussion by ghost** [SchneiderInfosystems/FB4D#72] [Q&A] [answered]
Could you help me
how to JSONObject.SetStringValue('testing') create based on ListBox1.ItemByIndex (c).item.itemdata.text;
for i :=0 to ListBox1.Count -1 do

Doc.AddOrUpdateField (TJSONObject.SetArray ('test'),[TJSONObject.SetStringValue('testing');‏‏‎

**Comment by SchneiderInfosystems** [comment]
Sorry for the late answer.

For me your question is unclear. First of all, you should make a plan how you want to organize your data in Firestore. Do you want to put all data in one document or should each ListItem have its own document? For second solution check out the new FSSimpleChat example, it does exactly that.

If this doesn't answer your question, don't hesitate to provide more information about what exactly you want to store and how to access the data.

Christoph

**Comment by SchneiderInfosystems** [comment, marked as answer]
Please open always a new discussion thread for a new topic!

Within the Intro Demo Application, you find in `TfmxFirebaseDemo.btnUploadAsynchClick` an example of how to upload an image file.

**Reply by SchneiderInfosystems** [reply]
Check the new discussion thread here: Example Code for download a picture from Android into the Storage

**Reply by SchneiderInfosystems** [reply]
Pls change to the right discussion thread and explain your problem!
